generators/client-2/templates/src/main/webapp/app/home/_home.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { Account } from "../shared/account.model";
+
+@Component({
+    selector: 'home',
+    templateUrl: 'app/home/home.html'
+})
+export class HomeComponent implements OnInit {
+    account: Account;
+    login: Function;
+    isAuthenticated: Function;
+
+    constructor(@Inject('Principal') private principal,
+                @Inject('$state') private $state,
+                @Inject('LoginService') loginService) {
+        this.login = loginService.open;
+    }
+
+    ngOnInit() {
+        //TODO: Remove this line after migrating "Principal" service
+        let that = this;
+        this.principal.identity().then(function (account) {
+            that.account = account;
+            that.isAuthenticated = that.principal.isAuthenticated;
+        });
```

Comment:
you can also do this, anyway not trivial since we need to change this part anyways so im merging it
```
        this.principal.identity().then(function (account) {
             this.account = account;
             this.isAuthenticated = this.principal.isAuthenticated;
         }.bind(this));
```